<?xml version="1.0"?>
<launch>
  <node name="joy" pkg="joy" type="joy_node" output="screen">
    <param name="default_trig_val" value="True" />
  </node>
  <node name="joystick_twist" pkg="e2e_handover" type="joystick_twist.py" output="screen"/>
  <node name="twist_filter" pkg="e2e_handover" type="twist_filter.py" output="screen">
    <remap from="/twist_filter/twist_cmd_raw"    to="/twist_cmd_smoothed"/>
    <remap from="/twist_filter/twist_cmd_filtered" to="/twist_controller/command" />
  </node>
  <node name="velocity_smoother" pkg="e2e_handover" type="velocity_smoother" output="screen">
    <rosparam file="$(find e2e_handover)/param/velocity_smoother.yaml" command="load"/>
    <remap from="velocity_smoother/raw_cmd_vel"    to="/twist_cmd_raw"/>
    <remap from="velocity_smoother/smooth_cmd_vel" to="/twist_cmd_smoothed"/>
  </node>
  <include file="$(find e2e_handover)/launch/ur5_bringup.launch" pass_all_args="true">
    <arg name="robot_ip" value="10.0.0.2"/>
  </include>
</launch>